test/test_core/test_tree_policies/test_UCT_TreePolicy.cpp

```diff
@@ -0,0 +1,111 @@
+#include <UCT_TreePolicy.h>
+#include <gtest/gtest.h>
+#include <memory>
+#include <string>
+
+State tempState(std::nullopt);
+
+class UCT_TreePolicyTest : public ::testing::Test {
+
+  protected:
+    void SetUp() override {
+
+        // root
+        State rootNode = State("rootNode");
+        root = SearchNode::create_SearchNode(nullptr, rootNode, false);
+
+        // root as parent
+        State rootChild1State = State("rootChild1");
+        State rootChild2State = State("rootChild2");
+        std::shared_ptr<SearchNode> rootChild1 = SearchNode::create_SearchNode(root, rootChild1State, false);
+        std::shared_ptr<SearchNode> rootChild2 = SearchNode::create_SearchNode(root, rootChild2State, false);
+
+        std::function<std::shared_ptr<SearchNode>(std::shared_ptr<SearchNode>)> sExp =
+            std::bind(&UCT_TreePolicyTest::simpleExpand, this, std::placeholders::_1);
+        std::function<std::shared_ptr<SearchNode>(std::shared_ptr<SearchNode>)> sChild =
+            std::bind(&UCT_TreePolicyTest::simpleChild, this, std::placeholders::_1);
+        treePolicy = UCT_TreePolicy(sExp, sChild);
+    }
+
+    std::shared_ptr<SearchNode> root = SearchNode::create_SearchNode(nullptr, tempState, false);
+    std::function<std::shared_ptr<SearchNode>(std::shared_ptr<SearchNode> node)> placeholderFunc = [](std::shared_ptr<SearchNode> node) {
```

Comment:
nice 139 char line

Reply:
What does the `[]` syntax mean?

Reply:
[] is the capture specification for lambda expressions

Reply:
Ah of course, I forgot that syntax, looks weird with empty capture body